### PR TITLE
Update Change log for V0.8

### PR DESCRIPTION
**Pull request by radhikaj**
Signed-off-by: radhikaj <radhikaj@microsoft.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -260,10 +276,19 @@ as listed below.
 
 Initial private preview release, no longer supported.
 
-[Unreleased](https://github.com/openenclave/openenclave/compare/v0.7.0...HEAD)
+[Unreleased](https://github.com/openenclave/openenclave/compare/v0.8.0...HEAD)
+
+[v0.8.0](https://github.com/openenclave/openenclave/compare/v0.8.0...v0.7.0)
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: The range is inverted, should be `v0.7.0...v0.8.0`

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,11 +10,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+[v0.8.0] - 2020-01-22
+---------------------
+
 ### Added
 
 - Support for backtracing in debug and release builds.
     - Implementations for GNU functions `backtrace` and `backtrace_symbols` (defined in execinfo.h)
     - Enclaves are built using `-fno-omit-frame-pointer` for accurate backtraces.
+- Support for custom attestation data formats via new plugin model.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: This probably requires more elaboration, can we link to their sample/docs on this? If they don't already, they need to add them, and if they do, we need to check that they are up-to-date with what is in the code.

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - The copyright for all sources is now attributed to Open Enclave SDK contributors.
 - Update Intel DCAP library dependencies to 1.4.1.
 - Update Intel PSW dependencies to 2.6.100.2 on Windows.
-- The stack guard for OE SDK on SGX is turned on. It is also turned on for enclaves.
+- Enable /W2 to treat W2 warnings as errors for Windows builds. Additionally, fix errors which arise from enabling this flag.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Technically it's the combination of `/W2` and `/WX` to treat level two warnings as errors. I would reword this so that it won't be misread as requiring any action on the app dev's part:
```
Enable `/W2 /WX` on Windows builds by default to treat W2 warnings as errors.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fix oe_random for size > 1024.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Keepachangelog.com guidance is to have a separate __Fixed__ section for bug fixes of note (see previous version notes in this file)

```
### Fixed
- `oe_random()` now correctly returns a fully filled byte buffer for requests of > 1024 bytes.
- Add `openenclave` namespace to dl and crypto libraries to prevent symbol collisions. Fixes #2082. 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Add openenclave namespace to dl and crypto libraries. Fixes #2082
+- Implement host side sockets on Windows.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: I would put this in the **Added** section

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Removed code related to deprecation of strftime.
+- Stack guard for enclaves is enabled. Enclave libs and enclaves are built using `-fstack-protector-strong`.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Consider emphasizing the default nature of the new setting:
```
- Build OE enclave libraries with stack protector enabled.
    - Enable `-fstack-protector-strong` by default for enclave application build configurations in cmake and pkgconfig.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Enclave libs and enclaves are built using `-gc-sections`.
+- Overhaul OCaml build system to use esy and upgraded OCaml version with reproducibility.
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: I would rephrase this in terms of impact to the user, which is that the OE SDK project now has a new set of dependencies that a contributor would need to refresh their setup with. Some of @andschwa instructions might be useful here.

File: CHANGELOG.md [hunk 8]
```diff
@@ -26,14 +30,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Bump ansible from 2.8.0 to 2.8.2 in /scripts/ansible
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Nit: I would replace "Bump" for "Update" for consistent word choice, and also pay attention to capitalization of product names:

```
- Update Ansible dependency from 2.8.0 to 2.8.2 in /scripts/ansible.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -260,10 +276,19 @@ as listed below.
 
 Initial private preview release, no longer supported.
 
-[Unreleased](https://github.com/openenclave/openenclave/compare/v0.7.0...HEAD)
+[Unreleased](https://github.com/openenclave/openenclave/compare/v0.8.0...HEAD)
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: The syntax here should changed so that the links are in the headers:
```
[Unreleased]: https://github.com/openenclave/openenclave/compare/v0.8.0...HEAD
[v0.8.0]: https://github.com/openenclave/openenclave/compare/v0.7.0...v0.8.0)

...
```

File: CHANGELOG.md (hunk 2 above)
Comment (radhikaj) [reply CHANGELOG.md]: Added a link to design doc.

File: CHANGELOG.md (hunk 9 above)
Comment (radhikaj) [reply CHANGELOG.md]: Sorry, I dont underst understand: so that links are in the headers? Which headers?

Comment (CodeMonkeyLeet) [reply CHANGELOG.md]: See https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links. we should be using reference-style links, which is why all the headers are in braces, e.g. 
```
[Unreleased]
------------
```

File: CHANGELOG.md (hunk 7 above)
Comment (andschwa) [reply CHANGELOG.md]: I don't think we've been focusing on "impact to user" in this changelog. For instance, several of the above changelog items are of a similar nature to this one.

We're bumping a new version; I would expect devs of the project to expect breaking changes. However, it's only going to apply if they've been building from source, but only from the last release branch, not from master. The instructions to build and run should be sufficient. (We could call out that they will want to review the build instructions if they're building from source and haven't pulled since the last release.)

Comment (radhikaj) [reply CHANGELOG.md]: @CodeMonkeyLeet @andschwa 
This is the wording I am considering:
- Replace OCPWin and OCaml with esy. T
   - The CMake-driven OCaml build is replaced with esy and dune. To install esy(which is now a prerequisite to building the SDK from source):
      - On Linux, `sudo ansible-playbook oe-linux-esy-setup.yml`
      - On Windows, `npm install -g esy@0.5.8`

Comment (radhikaj) [reply CHANGELOG.md]: This is the PR I referenced: https://github.com/openenclave/openenclave/pull/2221

File: CHANGELOG.md [hunk 10]
```diff
@@ -10,11 +10,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+[v0.8.0] - 2020-01-22
+---------------------
+
 ### Added
 
 - Support for backtracing in debug and release builds.
     - Implementations for GNU functions `backtrace` and `backtrace_symbols` (defined in execinfo.h)
     - Enclaves are built using `-fno-omit-frame-pointer` for accurate backtraces.
+- Support for custom attestation data formats via new plugin model. Please refer to [design documentation](docs/DesignDocs/CustomAttestation.md).
+- Support for host side sockets on Windows.
```
Comment (anakrish) [inline CHANGELOG.md]: There seems to be no mention of sockets for linux elsewhere. Are sockets an internal feature?

File: CHANGELOG.md (hunk 9 above)
Comment (radhikaj) [reply CHANGELOG.md]: Got it. Done. Thanks @CodeMonkeyLeet

File: CHANGELOG.md (hunk 10 above)
Comment (radhikaj) [reply CHANGELOG.md]: Sockets are not internal as enclave apps can call call the POSIX socket APIs. You are right on Linux support not being mentioned. It perhaps was an oversight.

File: CHANGELOG.md [hunk 11]
```diff
@@ -26,15 +31,30 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - The copyright for all sources is now attributed to Open Enclave SDK contributors.
 - Update Intel DCAP library dependencies to 1.4.1.
 - Update Intel PSW dependencies to 2.6.100.2 on Windows.
-- The stack guard for OE SDK on SGX is turned on. It is also turned on for enclaves.
+- Enable `/W2 /WX` on Windows builds by default to treat W2 warnings as errors.
+- Removed code related to deprecation of strftime.
+- Build OE enclave libraries with stack protector enabled.
```
Comment (anakrish) [inline CHANGELOG.md]: Should this be in added since this is a new feature we are supporting?

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,14 +7,19 @@ breaking changes should be noted here. It should be more concise than `git log`.
...
 
+[v0.8.0][v0.8.0_log] - 2020-01-22
+---------------------
+
 ### Added
 
 - Support for backtracing in debug and release builds.
     - Implementations for GNU functions `backtrace` and `backtrace_symbols` (defined in execinfo.h)
     - Enclaves are built using `-fno-omit-frame-pointer` for accurate backtraces.
+- Support for custom attestation data formats via new plugin model. Please refer to [design documentation](docs/DesignDocs/CustomAttestation.md).
```
Comment (CodeMonkeyLeet) [inline CHANGELOG.md]: Grammar (missing article): "Please refer to _the_ design documentation."

Can we file an issue to track ensuring that the doc actually reflects the current implementation if we are going to use it as public documentation? A quick scan shows that some of the type names have changed at least (e.g. `oe_attester_plugin_t` is now `oe_attestation_role_t`)

File: CHANGELOG.md (hunk 11 above)
Comment (radhikaj) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 12 above)
Comment (radhikaj) [reply CHANGELOG.md]: Fixed. thanks.

Comment (radhikaj) [reply CHANGELOG.md]: Opened issue #2458

**Comment by radhikaj** [comment]
Thanks @anakrish @CodeMonkeyLeet for reviewing.
